File: manuelalex/src/statements/IFStatement.js
/**
 * Created by Manuel on 13/02/2017.
 */

import {Statement}  from './Statement.js';

export class IfStatement extends Statement {

    constructor(condition, ifBody, location) {
        super(location);

        this.condition = condition;
        this.ifBody = ifBody;

    }

    getCondition(){
        return this.condition;
    }

    getIfBody(){
        return this.ifBody;
    }

    accept(visitor){
        visitor.visitIfStatement(this);
    }

    render(visitor, view){
        visitor.renderIfStatement(this, view);
    }
}

File: koopmans/lib/ql/ast/statement/if_statement.rb
module QL
  module AST
    class IfStatement
      attr_reader :body, :condition

      def initialize(condition, body)
        @condition = condition
        @body = body
      end

      def accept(visitor, argument = nil)
        visitor.visit_if_statement(self, argument)
      end
    end
  end
end
